cmd/catalyst-node/handlers/access-control/access-control.go

```diff
@@ -134,36 +132,73 @@ func (ac *PlaybackAccessControl) handleUserNew(payload []byte) []byte {
 	return []byte("false")
 }
 
-func (ac *PlaybackAccessControl) getPlaybackAccessControlInfo(playbackID, pubKey string) (bool, error) {
+var playbackRequestPath = regexp.MustCompile(`hls/([^/]+)/`)
+
+func (ac *PlaybackAccessControl) IsAuthorized(reqURL *url.URL) (bool, error) {
+	matches := playbackRequestPath.FindStringSubmatch(reqURL.Path)
+	if len(matches) < 2 {
+		return false, fmt.Errorf("error extracting playbackID from request path %s", reqURL.Path)
+	}
+
+	acReq := PlaybackAccessControlRequest{Stream: matches[1]}
+	cacheKey := ""
+	accessKey := reqURL.Query().Get("accessKey")
+	jwt := reqURL.Query().Get("jwt")
+	if accessKey != "" {
+		acReq.Type = "accessKey"
+		acReq.AccessKey = accessKey
+		cacheKey = accessKey
+	} else if jwt != "" {
+		acReq.Pub = extractKeyFromJwt(jwt, acReq.Stream)
+		if acReq.Pub == "" {
+			return false, fmt.Errorf("failed to extract key from jwt: %s", jwt)
+		}
+
+		acReq.Type = "jwt"
+		cacheKey = acReq.Pub
```

Comment:
WDYT of adding a little prefix to these cache keys just to avoid potential conflicts? Specifically the `accessKey` value is 100% user-input, so they could end up providing something that conflicts with an existing pub key I guess.

```suggestion
		cacheKey = "accessKey_" + accessKey
	} else if jwt != "" {
		acReq.Pub = extractKeyFromJwt(jwt, acReq.Stream)
		if acReq.Pub == "" {
			return false, fmt.Errorf("failed to extract key from jwt: %s", jwt)
		}

		acReq.Type = "jwt"
		cacheKey = "jwtPubKey_" + acReq.Pub
```

Reply:
Omg there would actually be a very real attack here: just send an `accessKey` for a resource that is gated via JWT, with the same value as the JWT public key, then you'd get access to the stream without a real JWT :o

Reply:
oh wow yeah good spot